fbw-a380x/src/systems/systems-host/systems/LegacyGpws.ts

```diff
@@ -181,33 +181,52 @@ export class LegacyGpws {
+        const computedAirspeed = Arinc429Word.fromSimVarValue("L:A32NX_ADIRS_ADR_1_COMPUTED_AIRSPEED");
+        const pitch = Arinc429Word.fromSimVarValue("L:A32NX_ADIRS_IR_1_PITCH");
+        const inertialVs = Arinc429Word.fromSimVarValue("L:A32NX_ADIRS_IR_1_VERTICAL_SPEED");
+        const barometricVs = Arinc429Word.fromSimVarValue("L:A32NX_ADIRS_ADR_1_BAROMETRIC_VERTICAL_SPEED");
+        const radioAlt1 = Arinc429Word.fromSimVarValue("L:A32NX_RA_1_RADIO_ALTITUDE");
+        const radioAlt2 = Arinc429Word.fromSimVarValue("L:A32NX_RA_2_RADIO_ALTITUDE");
         const radioAlt = radioAlt1.isFailureWarning() || radioAlt1.isNoComputedData() ? radioAlt2 : radioAlt1;
         const radioAltValid = radioAlt.isNormalOperation();
-        const onGround = SimVar.GetSimVarValue('SIM ON GROUND', 'Bool');
+        const isOnGround = !this.isAirVsGroundMode;
 
+        const isGpwsSysOff = SimVar.GetSimVarValue("L:A32NX_GPWS_SYS_OFF", "Bool") === 1;
+        const isTerrModeOff = SimVar.GetSimVarValue("L:A32NX_GPWS_TERR_OFF", "Bool") === 1;
+        const isFlapModeOff = SimVar.GetSimVarValue("L:A32NX_GPWS_FLAP_OFF", "Bool") === 1;
+
+        const sfccPositionWord = Arinc429Word.fromSimVarValue("L:A32NX_SFCC_SLAT_FLAP_ACTUAL_POSITION_WORD");
+        const isFlapsFull = sfccPositionWord.getBitValueOr(22, false);
+
+        const isGearDownLocked = SimVar.GetSimVarValue("L:A32NX_LGCIU_1_LEFT_GEAR_DOWNLOCKED", "Bool") === 1;
+
+        // TODO only use this in the air?
+        const isNavAccuracyHigh = SimVar.GetSimVarValue("L:A32NX_FMGC_L_NAV_ACCURACY_HIGH", "Bool") === 1;
+        const isTcfOperational = this.isTerrainClearanceFloorOperational(isTerrModeOff, radioAlt, isNavAccuracyHigh);
+        const isTafOperational = this.isTerrainAwarenessOperational(isTerrModeOff);
         this.UpdateAltState(radioAltValid ? radioAlt.value : NaN);
         this.differentiateRadioalt(radioAltValid ? radioAlt.value : NaN, deltaTime);
 
         const mda = SimVar.GetSimVarValue('L:AIRLINER_MINIMUM_DESCENT_ALTITUDE', 'feet');
         const dh = SimVar.GetSimVarValue('L:AIRLINER_DECISION_HEIGHT', 'feet');
         const phase = SimVar.GetSimVarValue('L:A32NX_FMGC_FLIGHT_PHASE', 'Enum');
 
+      this.update_maxRA(radioAlt, isOnGround);
+
+        const isOverflightDetected = this.updateOverflightState(deltaTime);
+        this.updateMode4aInhibited(isGearDownLocked, areFlapsInLandingConfig);
+
         if (
             radioAltValid && radioAlt.value >= 10 && radioAlt.value <= 2450
             && !SimVar.GetSimVarValue('L:A32NX_GPWS_SYS_OFF', 'Bool')
         ) { // Activate between 10 - 2450 radio alt unless SYS is off
-            const FlapPushButton = SimVar.GetSimVarValue('L:A32NX_GPWS_FLAPS3', 'Bool');
             const FlapPosition = SimVar.GetSimVarValue('L:A32NX_FLAPS_HANDLE_INDEX', 'Number');
             const FlapsInLandingConfig = FlapPushButton ? (FlapPosition === 3) : (FlapPosition === 4);
```

Comment:
But does here (with a different name). If you hook the "FlapPushButton" up to whatever data we get that from on the A380 and make it available, then you've got flaps 3 sorted.

Reply:
I connected the areFlapsInLandingConfig constant with the FlapsInLandingConfig constant.

Reply:
Just now.